### PR TITLE
[API] Allow certain endpoints to return JSON

### PR DESCRIPTION
**Pull request by astronaut-wannabe**
# How does this pull request make you feel (in animated GIF format)?

MY B

# What are the relevant GitHub issues?

relates to #1235 

# What does this pull request do?

In [Pull Request #1418][0] I blocked ALL json requests

We actually have a few endpoints we want to let through

[0]:https://github.com/crimethinc/website/pull/1418

# How should this be manually tested?

- make sure the live polling articles still work
- https://crimethinc-staging-pr-1427.herokuapp.com/manifest.json works
- https://crimethinc-staging-pr-1427.herokuapp.com/feed.json works
  - https://crimethinc-staging-pr-1427.herokuapp.com/feed.json?page=3 works

# Questions for the pull request author (delete any that don't apply):
- [X] Does the code you updated have tests? If not, could you add some please?

# Acceptance Criteria
## These should be checked by the reviewers

- [x] This pull request does not cause the database export script to become out of sync with the db schema

### CODE REVIEW

File: app/middlewares/rack/json_requests.rb [hunk 1]
```diff
@@ -12,14 +12,20 @@ def initialize app
...
-      return json_406_error if content_type.start_with?('application/json')
+      return json_406_error if invalid_api_request? request
 
       @app.call(env)
     end
 
     private
 
+    def invalid_api_request? request
+      return false if request.path.include?('collection_posts')
```
Comment (astronaut-wannabe) [inline app/middlewares/rack/json_requests.rb]: are there other paths?

File: app/middlewares/rack/json_requests.rb [hunk 2]
```diff
@@ -12,14 +12,21 @@ def initialize app
...
+      return json_406_error if invalid_api_request? request
 
       @app.call(env)
     end
 
     private
 
+    def invalid_api_request? request
+      return false if request.path.include?('collection_posts') ||
+                      request.path.include?('manifest.json')
```
Comment (astronaut-wannabe) [inline app/middlewares/rack/json_requests.rb]: are there other paths?

File: app/middlewares/rack/json_requests.rb [hunk 3]
```diff
@@ -12,14 +12,22 @@ def initialize app
...
 
       @app.call(env)
     end
 
     private
 
+    def invalid_api_request? request
+      return false if request.path.include?('collection_posts') ||
+                      request.path.include?('manifest.json') ||
+                      request.path.include?('feed.json')
```
Comment (astronaut-wannabe) [inline app/middlewares/rack/json_requests.rb]: are there other paths?

Comment (veganstraightedge) [reply app/middlewares/rack/json_requests.rb]: Not that I can think of. Bugsnag will tell us if we're wrong.

**Pull request merged**